pkg/storage/s3/s3.go

```diff
@@ -190,6 +204,7 @@ func (d *driver) ConfigEnv() (envs envvar.List, err error) {
 		envvar.EnvVar{Name: "REGISTRY_STORAGE_S3_BUCKET", Value: d.Config.Bucket},
 		envvar.EnvVar{Name: "REGISTRY_STORAGE_S3_REGION", Value: d.Config.Region},
 		envvar.EnvVar{Name: "REGISTRY_STORAGE_S3_ENCRYPT", Value: d.Config.Encrypt},
+		envvar.EnvVar{Name: "REGISTRY_STORAGE_S3_VIRTUALHOSTEDSTYLE", Value: d.Config.VirtualHostedStyle},
```

Comment:
Do we want to always set this env var, or only set it if the custom region endpoint is set?

Reply:
It shouldn't hurt anything, but we will need to make sure that the documentation details that it only works with the custom region endpoint.

Reply:
Interesting point, we need to update https://github.com/docker/distribution/blob/master/docs/configuration.md in our upstream PR and eventually https://github.com/docker/docker.github.io/blob/master/registry/storage-drivers/s3.md